raft/server.cc

```diff
@@ -1179,8 +1180,11 @@ future<> server_impl::abort() {
         f.set_exception(std::runtime_error("Snapshot application aborted"));
     }
 
-    co_await seastar::when_all_succeed(std::move(abort_rpc), std::move(abort_sm), std::move(abort_persistence)).discard_result();
-
+    // Destroy entry waiters before waiting for `abort_rpc`,
+    // since the RPC implementation may wait for forwarded `modify_config` calls to finish
```

Comment:
Shouldn't this be folded into the previous patch since before adding the gate around the rpc the problem did not exist?

Reply:
Makes sense, I can squash the two.